xarray/backends/api.py

```diff
@@ -198,6 +198,9 @@ def check_attr(name, value):
                 "serialization to netCDF files".format(name)
             )
 
+        if isinstance(value, dict) and ("zarr" in backend):
```

Comment:
Hmm.. This looks like we are inventing a convention to store nested dictionaries; is that right?

Reply:
I thought the same, but it is not the case. The nested dictionary is stored correctly in the .zattrs file, and by correctly I mean in the same way that using Zarr would store them. If we use Zarr directly to read the attributes I get a nested dictionary.